### DISCUSSION TITLE
FreeRDP dependency in weston project for WSLg

### DISCUSSION THREAD

**Discussion by hideyukn88** [FreeRDP/FreeRDP#7395] [Q&A]
Hi all,

As upstreaming from WSLg to FreeRDP master is now [completed ](https://github.com/FreeRDP/FreeRDP/pull/7381)(thank you for help!), we are looking for upstreaming our changes in weston compositor next. This is where we need advise from FreeRDP community. Currently weston project, particularly, its RDP-backend, [expects FreeRDP 2.2.0](https://github.com/wayland-project/weston/blob/main/libweston/backend-rdp/meson.build#L7), and WSLg is now built with FreeRDP 2.4.0 plus WSLg changes at our forked branch. Thus, when upstreaming our weston changes, we have 2 possible options for FreeRDP dependency in weston project.

1: WSLg changes in weston will only be enabled for FreeRDP 3.0.0 or greater (assuming the latest of master will be used).
2: Request backport of WSLg changes to stable-2.0, and wait for that to happen, then make it FreeRDP 2.x.x or greater dependent.

So here is our questions,

1: Is there any plan for stable-3.0 (with our WSLg change included)? if so, any estimated timeframe?

2: (while we didn’t do that in our upstream PR, but) should we bump up version (say to 3.0.1) so it allows weston project to determine whether paired FreeRDP 3.x.x source has WSLg changes or not (of course, meson can check by header presence or many other ways, but I think version is always the most explicit way to tell as whole).

3: If no immediate plan for 3.0, if there any process to request backport WSLg changes to 2.0? Can this type of change meet criteria of backport?

@awakecoding @hardening @bmiklautz @akallabeth, thanks for reviewing and approving changes, and any advise on this topic is greatly appreciated.

Thank you very much for all your help!

**Comment by hardening** [comment]
Hi, sorry for the very late answer
1. AFAICT we don't have a very precise release plan, 3.0 will probably be frozen when we will have enough new features in it. We kinda had in mind a yearly release idea for some time, but we have been taken back to reality, and given the agenda of the various people working on FreeRDP that looks too short.
2. we have not been very careful about API compatibility before FreeRDP 2.0, but now we take a special attention to not break public APIs
3. we usually only backport fixes, or very localized things (like for example adding the Android decoder which is a limited local change). The point for backports is probably to not break API and not enter again in a cycle where FreeRDP 2.0 would have a different API compared to FreeRDP 2.8 and client code has to deal with this (in horrible ways)

That's only my view, not necessary shared by other FreeRDP members
